### DISCUSSION TITLE
rewrites , How does route rewriting implement such a scenario

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#31834] [Help]
### Describe the feature you'd like to request

module.exports = {
  async rewrites() {
    return [
      {
        source: '/old-about/:path*', // The access path of my two pages is through one address
        destination: '/about', 
      },
      {
        source: '/old-about/:path*', // The access path of my two pages is through one address
        destination: '/about-details',  //  But the place I want to go is different
      },
    ]
  },
}

The following questions: 

as follows:   https://www.abc.com/old-about/74883873.html，   Want to rewrite to   /about
as follows:  https://www.abc.com/old-about/a24b12，   Want to rewrite to   /about-details

I can't write regular to match suffixes ??? 
as follows:    It would be perfect if I could match suffixes and achieve different mappings
{
        source: '/old-blog/:post(\\d{1,})',
        destination: '/blog/:post', // Matched parameters can be used in the destination
 },

### Describe the solution you'd like

The following questions: 

as follows:   https://www.abc.com/old-about/74883873.html，   Want to rewrite to   /about
as follows:  https://www.abc.com/old-about/a24b12，   Want to rewrite to   /about-details

I can't write regular to match suffixes ??? 
as follows:    It would be perfect if I could match suffixes and achieve different mappings
{
        source: '/old-blog/:post(\\d{1,})',
        destination: '/blog/:post', // Matched parameters can be used in the destination
 },

### Describe alternatives you've considered

The following questions: 

as follows:   https://www.abc.com/old-about/74883873.html，   Want to rewrite to   /about
as follows:  https://www.abc.com/old-about/a24b12，   Want to rewrite to   /about-details

I can't write regular to match suffixes ??? 
as follows:    It would be perfect if I could match suffixes and achieve different mappings
{
        source: '/old-blog/:post(\\d{1,})',
        destination: '/blog/:post', // Matched parameters can be used in the destination
 },

**Comment by parlay96** [comment]
Because I must be compatible with previous website access links， I now refactor the previous project using next

**Comment by parlay96** [comment]
？

**Comment by shuding** [comment]
You can match paths with Regex: https://nextjs.org/docs/api-reference/next.config.js/rewrites#regex-path-matching.

**Comment by parlay96** [comment]
I tried. It didn't work

**Comment by parlay96** [comment]
path: '/resume-details/:slug(/^\d+.html/)',   Why is it invalid
path: '/resume-details/:slug(\d+.html)',  Is that right?
 Why remove    /^ ？

**Comment by parlay96** [comment]
path: '/resume-details/:slug((\\d+).html)'   No effect

**Comment by parlay96** [comment]
![image](https://user-images.githubusercontent.com/32978177/143803499-42fe2d35-31e6-49a7-ae71-033e6a7cb62a.png)
If I wrap it up???

**Comment by parlay96** [comment]
What do I need to do

**Comment by parlay96** [comment]
path: "/zhaohuo/:slug([acs]\\d+.*\/?[recommend|newest|auth|cert|project]?)",

I want to be like this： path: "/zhaohuo/:slug( ([acs]\\d+.*）\/?（recommend|newest|auth|cert|project）?)",

But it will report an error

**Comment by parlay96** [comment]
.

**Comment by parlay96** [comment]
Can you answer it? Thank you
